vertical-pod-autoscaler/pkg/utils/metrics/quality/quality.go

```diff
@@ -44,15 +53,51 @@ var (
 		prometheus.CounterOpts{
 			Namespace: metricsNamespace,
 			Name:      "usage_sample_missing_recommendation_count",
-			Help:      "Count of usage samples when a recommendation should be present but is missing.",
+			Help:      "Count of usage samples when a recommendation should be present but is missing",
 		}, []string{"resource", "is_oom"},
 	)
+	cpuUsageRecommendationDiff = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "cpu_usage_recommendation_diffs_cores",
+			Help:      "Absolute diffs between usage and recommendation for cpu",
+			Buckets:   prependNegativeBuckets(cpuBuckets),
+		},
+	)
+	memoryUsageRecommendationDiff = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "mem_usage_recommendation_diffs_bytes",
+			Help:      "Absolute diffs between usage and recommendation for memory",
+			Buckets:   prependNegativeBuckets(memoryBuckets),
+		}, []string{"is_oom"},
+	)
+	cpuRecommendations = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "cpu_recommendations_cores",
+			Help:      "CPU recommendation values as observed on recorded usage sample",
+			Buckets:   cpuBuckets,
+		},
+	)
+	memoryRecommendations = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "mem_recommendations_bytes",
+			Help:      "Memory recommendation values as observed on recorded usage sample",
+			Buckets:   memoryBuckets,
+		}, []string{"is_oom"},
```

Comment:
Sorry for bringing it up only now and not in the previous PR - but do you think it would make sense to add "VPA mode" as a label to all these new metrics?

Reply:
No problem. Added.